pkg/controller/installation/installation_controller.go

```diff
@@ -377,12 +391,265 @@ func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.
 		return retryRequeue, nil
 	}
 	installation.Status.Stage = integreatlyv1alpha1.StageName("complete")
+
+	if (isFirstInstallReconcile(installation) || isUpgradeReconcile(installation)) && allProductsReconciled(installation) {
+		installation.Status.Version = installation.Status.ToVersion
+		installation.Status.ToVersion = ""
+		metrics.SetRhmiVersions(string(installation.Status.Stage), installation.Status.Version, installation.Status.ToVersion)
+	}
+
 	_ = r.client.Status().Update(r.context, installation)
 	metrics.RHMIStatusAvailable.Set(1)
 	logrus.Infof("installation completed succesfully")
 	return reconcile.Result{Requeue: true, RequeueAfter: 5 * time.Minute}, nil
 }
 
+func isFirstInstallReconcile(installation *integreatlyv1alpha1.RHMI) bool {
+	return installation.Status.Version == ""
+}
+
+func isUpgradeReconcile(installation *integreatlyv1alpha1.RHMI) bool {
+	status := installation.Status
+	if status.ToVersion != "" {
+		return true
+	}
+	return false
+}
+
+func allProductsReconciled(installation *integreatlyv1alpha1.RHMI) bool {
+	stages := installation.Status.Stages
+
+	return (verifyClusterRHSSOVersions(stages) &&
+		verifyCloudResourceVersions(stages) &&
+		verifyAMQOnlineVersions(stages) &&
+		verifyApicuritoVersions(stages) &&
+		verifyCodeReadyWorkspacesVersions(stages) &&
+		verifyFuseOnOpenshiftVersions(stages) &&
+		verifyMonitoringVersions(stages) &&
+		verify3ScaleVersions(stages) &&
+		verifyFuseOnlineVersions(stages) &&
+		verifyDataSyncVersions(stages) &&
+		verifyRHSSOUserVersions(stages) &&
+		verifyUPSVersions(stages) &&
+		verifySolutionExplorerVersions(stages))
+}
```

Comment:
Just a thought, would we be better getting out of this check as soon as possible. If any product verify function returns a  false  the whole function output will be a false. 
```
if !verifySolutionExplorerVersions(stages){
    return false
}
if !verifyUPSVersions(stages) {
    return false
}
return true
```

Reply:
this is called Short circuit evaluation, it will get out as sooner one of the function return false

 https://kuree.gitbooks.io/the-go-programming-language-report/content/22/text.html

Reply:
That's good to hear, and thanks for the link too. It looks like there maybe a few other answers in there to questions I have.